### PR TITLE
libcap => 2.62

### PR DESCRIPTION
**Pull request by saltedcoffii**
libcap upgrade 2.62. works on x86_64. needs binaries.

##### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/saltedcoffii/chromebrew.git CREW_TESTING_BRANCH=libcap_2.62 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/libcap.rb [hunk 1]
```diff
@@ -29,21 +17,25 @@ class Libcap < Package
...
     # change the path to ld
     system "sed -i 's,/usr/bin/ld,#{CREW_PREFIX}/bin/ld,g' Make.Rules"
     # change prefix
-    system "sed -i 's,prefix=/usr,prefix=#{CREW_PREFIX},' Make.Rules"
+    system "sed -i 's,prefix=/usr,prefix=#{CREW_PREFIX},g' Make.Rules"
     # set exec_prefix
-    system "sed -i 's,^exec_prefix=,exec_prefix=\$(prefix),' Make.Rules"
+    system "sed -i 's,^exec_prefix=,exec_prefix=\$(prefix),g' Make.Rules"
+    # use system libdir
+    system "sed -i 's,^lib_prefix=$(exec_prefix),lib_prefix=#{CREW_LIB_PREFIX},g' Make.Rules"
```
Comment (uberhacker) [inline packages/libcap.rb:27]: Escape `$` with `\`?

**Comment by satmandu** [comment]
```
make[1]: Entering directory '/usr/local/tmp/crew/libcap.20211228210950.dir/go'
CC="gcc" CGO_LDFLAGS_ALLOW="-Wl,-?-wrap[=,][^-.@][^,]*" CGO_CFLAGS="-I/usr/local/tmp/crew/libcap.20211228210950.dir/libcap/include" CGO_LDFLAGS="-L/usr/local/tmp/crew/libcap.20211228210950.dir/libcap" go build  -mod=vendor compare-cap.go
CC="gcc" CGO_ENABLED=0 go build  -mod=vendor ok.go
CC="gcc" CGO_ENABLED="1" CGO_LDFLAGS_ALLOW="-Wl,-?-wrap[=,][^-.@][^,]*" go build  -mod=vendor try-launching.go
CC="gcc" CGO_ENABLED="1" CGO_LDFLAGS_ALLOW="-Wl,-?-wrap[=,][^-.@][^,]*" CGO_CFLAGS="-I/usr/local/tmp/crew/libcap.20211228210950.dir/libcap/include" CGO_LDFLAGS="-L/usr/local/tmp/crew/libcap.20211228210950.dir/libcap" go build  -mod=vendor psx-signals.go
CC="gcc" CGO_ENABLED="1" CGO_LDFLAGS_ALLOW="-Wl,-?-wrap[=,][^-.@][^,]*" go test -mod=vendor kernel.org/pub/linux/libs/security/libcap/psx
ok  	kernel.org/pub/linux/libs/security/libcap/psx	0.037s
CC="gcc" CGO_ENABLED="1" CGO_LDFLAGS_ALLOW="-Wl,-?-wrap[=,][^-.@][^,]*" go test -mod=vendor kernel.org/pub/linux/libs/security/libcap/cap
panic: test timed out after 10m0s

goroutine 15 [running]:

goroutine 1 [chan receive, 10 minutes]:
testing.T.Run
	:0
testing.M.Run
	:0
main.main
	/tmp/go-build/b001/_testmain.go:55

goroutine 13 [runnable]:
syscall.Exitsyscall
	:0
syscall.Syscall
	:0
syscall.Tgkill
	:0
kernel_0org_1pub_1linux_1libs_1security_1libcap_1cap.lResult.pollForThreadExit
	/usr/local/tmp/crew/libcap.20211228210950.dir/go/vendor/kernel.org/pub/linux/libs/security/libcap/cap/launch.go:366
kernel_0org_1pub_1linux_1libs_1security_1libcap_1cap.Launcher.Launch
	/usr/local/tmp/crew/libcap.20211228210950.dir/go/vendor/kernel.org/pub/linux/libs/security/libcap/cap/launch.go:421
kernel_0org_1pub_1linux_1libs_1security_1libcap_1cap.TestFuncLaunch
	/usr/local/tmp/crew/libcap.20211228210950.dir/go/vendor/kernel.org/pub/linux/libs/security/libcap/cap/cap_test.go:243
FAIL	kernel.org/pub/linux/libs/security/libcap/cap	600.079s
FAIL
make[1]: *** [Makefile:107: test] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/libcap.20211228210950.dir/go'
make: *** [Makefile:13: test] Error 2
libcap failed to build: `make test` exited with 2
```

**Comment by saltedcoffii** [comment]
i bet it has something to do with chromeos security issues as it built fine in my chroot. ugh.

**Comment by satmandu** [comment]
I built it in a container though...

**Comment by satmandu** [comment]
Hmmm. Maybe I have to disable the tests when building in a container?

**Comment by satmandu** [comment]
@saltedcoffii Please check if the generated binaries work and merge if they do...

File: packages/libcap.rb [hunk 2]
```diff
@@ -2,48 +2,55 @@
...
 
   def self.install
     system 'make', 'RAISE_SETFCAP=no', "DESTDIR=#{CREW_DEST_DIR}", 'install'
-    system "patchelf --set-rpath #{CREW_LIB_PREFIX} #{CREW_DEST_PREFIX}/sbin/capsh"
+  end
+
+  def self.check
+    @container_check = system 'grep :/docker /proc/self/cgroup &> /dev/null'
+    # Tests do not work in container.
+    return unless @container_check
```
Comment (uberhacker) [inline packages/libcap.rb:52]: `return if @container_check`?  What is returned?  True or false?

Comment (uberhacker) [reply packages/libcap.rb:52]: Do tests pass on actual hardware?

Comment (satmandu) [reply packages/libcap.rb:52]: > `return if @container_check`?  What is returned?  True or false?

`@container_check` is false if in a container.

Comment (satmandu) [reply packages/libcap.rb:52]: Actually that doesn't work. In docker the only reliable way I'm finding of determining if you are in a container is looking for `/.dockerenv` .

Comment (satmandu) [reply packages/libcap.rb]: Tests pass on `x86_64`, and checking now on `i686`... Do not have `armv7l` hardware to test on...

Comment (uberhacker) [reply packages/libcap.rb]: I can check on arm.  What are you doing to confirm your test?

Comment (satmandu) [reply packages/libcap.rb]: Passes on `i686` too.

Comment (uberhacker) [reply packages/libcap.rb]: That doesn't seem like a reliable way to check for a docker environment.  See https://tuhrig.de/how-to-know-you-are-inside-a-docker-container/.

Comment (satmandu) [reply packages/libcap.rb]: @supechicken FYI on container check in sommelier needing new logic...

Comment (satmandu) [reply packages/libcap.rb]: @uberhacker Can you check if the test also works on armv7l? I suspect it will...

Comment (satmandu) [reply packages/libcap.rb]: The method in your link doesn't work.

```
chronos@cheekon-x86_64 /proc/self $ cat /proc/1/cgroup
0::/
```

I used this method: https://stackoverflow.com/a/25518345

Comment (uberhacker) [reply packages/libcap.rb]: On armv7l, I get:
```
make -C doc all
make[1]: Entering directory '/usr/local/tmp/crew/libcap.20211228234705.dir/doc'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/usr/local/tmp/crew/libcap.20211228234705.dir/doc'
Preconfiguring package...
/usr/local/lib/crew/lib/package.rb:195:in `system': No such file or directory - make -j4
libcap failed to build: `make -j4 RAISE_SETFCAP=no DESTDIR=/usr/local/tmp/crew/dest install` exited with 127
```
Same inside the container.  I think there is a problem with @supechicken's latest crew update.

Comment (satmandu) [reply packages/libcap.rb]: Yeah you're hitting https://github.com/skycocker/chromebrew/issues/6529

Comment (uberhacker) [reply packages/libcap.rb]: That begs the question.  How have you been successfully generating the latest series of binaries?

**Comment by uberhacker** [comment]
Unfortunately, even after reverting @supechicken's changes, on the latest armv7l build attempt, I get the following:
```
ok      kernel.org/pub/linux/libs/security/libcap/psx   0.117s
CC="gcc" CGO_ENABLED="1" CGO_LDFLAGS_ALLOW="-Wl,-?-wrap[=,][^-.@][^,]*" go test -mod=vendor kernel.org/pub/linux/libs/security/libcap/cap
panic: test timed out after 10m0s

goroutine 19 [running]:

goroutine 1 [chan receive, 10 minutes]:
testing.T.Run
        :0
testing.M.Run
        :0
main.main
        /tmp/go-build/b001/_testmain.go:55

goroutine 13 [runnable]:
syscall.Exitsyscall
        :0
syscall.Syscall
        :0
syscall.Tgkill
        :0
kernel_0org_1pub_1linux_1libs_1security_1libcap_1cap.lResult.pollForThreadExit
        /usr/local/tmp/crew/libcap.20211228235802.dir/go/vendor/kernel.org/pub/linux/libs/security/libcap/cap/launch.go:366
kernel_0org_1pub_1linux_1libs_1security_1libcap_1cap.Launcher.Launch
        /usr/local/tmp/crew/libcap.20211228235802.dir/go/vendor/kernel.org/pub/linux/libs/security/libcap/cap/launch.go:421
kernel_0org_1pub_1linux_1libs_1security_1libcap_1cap.TestFuncLaunch
        /usr/local/tmp/crew/libcap.20211228235802.dir/go/vendor/kernel.org/pub/linux/libs/security/libcap/cap/cap_test.go:243
FAIL    kernel.org/pub/linux/libs/security/libcap/cap   600.154s
FAIL
make[1]: *** [Makefile:107: test] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/libcap.20211228235802.dir/go'
make: *** [Makefile:13: test] Error 2
libcap failed to build: `make test` exited with 2
```

**Comment by satmandu** [comment]
Does the captest binary I compiled for `armv7l` work?

**Comment by satmandu** [comment]
Whoops! `captest` is from `libcap_ng`...

**Comment by uberhacker** [comment]
Have the new binaries been built with the latest crew update?

**Comment by satmandu** [comment]
No, but they should be fine. Please check to see if the `armv7l` build tests are ok?

**Comment by uberhacker** [comment]
No and now I get yet another error with this new logic that always breaks the builds.
```
Rename all *.la files to *.la_tmp
/bin/sh: 1: Syntax error: "(" unexpected
/usr/local/lib/crew/lib/package.rb:174:in `system': Command failed with exit 2: bash -e -c "sed -e "/gperf --/s/gperf --/gperf --includes --/" -e "/gperf --/s/cap_lookup_name(/cap_dummy(/" -i libcap/Makefile"                                                                                                                                                                    
libcap failed to build: `sed -e "/gperf --/s/gperf --/gperf --includes --/" -e "/gperf --/s/cap_lookup_name(/cap_dummy(/" -i libcap/Makefile` exited with 2
```

**Comment by satmandu** [comment]
Try again after merging master...

**Pull request merged**